go/proof.go

```diff
@@ -309,6 +317,40 @@ func getPadding(spec *InnerSpec, branch int32) (minPrefix, maxPrefix, suffix int
 	return
 }
 
+// isPlaceholder returns true if the padding bytes correspond to all empty children
+// on a given side of this branch, ie. it's a valid placeholder for this position
+func isPlaceholder(spec *InnerSpec, branch int32, op *InnerOp, inPrefix bool) bool {
+	idx := getPosition(spec.ChildOrder, branch)
+
+	// compare the prefix bytes with the appropriate number of empty children
+	if inPrefix {
+		prefixCt := len(spec.ChildOrder) - 1 - idx
+		lenDiff := len(op.Prefix) - prefixCt*int(spec.ChildSize)
+		if lenDiff < int(spec.MinPrefixLength) || lenDiff > int(spec.MaxPrefixLength) {
+			return false
+		}
+		for i := 0; i < prefixCt; i++ {
```

Comment:
I get what you are trying to do, but I am unsure if this is correct.

Definitely could use some test cases... first a few hand-made examples just combining bytes and unit testing this method. Then a higher level on (in #57) that applies it to your real case.